com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -157,14 +148,14 @@ public UniversalRenderPipeline(UniversalRenderPipelineAsset asset)
             if (msaaSampleCountNeedsUpdate)
             {
                 QualitySettings.antiAliasing = asset.msaaSampleCount;
-#if ENABLE_VR && ENABLE_XR_MODULE
-                XRSystem.UpdateMSAALevel(asset.msaaSampleCount);
-#endif
             }
 
-#if ENABLE_VR && ENABLE_XR_MODULE
-            XRSystem.UpdateRenderScale(asset.renderScale);
-#endif
+
+            // Configure initial XR settings
+            MSAASamples msaaSamples = (MSAASamples)Mathf.Clamp(Mathf.NextPowerOfTwo(QualitySettings.antiAliasing), (int)MSAASamples.None, (int)MSAASamples.MSAA8x);
+            XRSystem.SetDisplayMSAASamples(msaaSamples);
+            XRSystem.SetRenderScale(asset.renderScale);
```

Comment:
XRSystem doesn't depend on the modules anymore I guess? That's nice.

Reply:
XR code stills needs the modules to be present but I moved the conditional compilation inside the core package to have a cleaner URP/HDRP integration.
If the modules are not present, the methods will be stripped out by the IL2CPP dead code removal.